packages/core/src/media/controls/pictureinpicture.ts

```diff
@@ -0,0 +1,116 @@
+import { isClient } from '../browser';
+
+const methodsList = [
+  // Modern Browsers
+  {
+    // request picture in picture
+    requestPictureInPicture: 'requestPictureInPicture',
+    // exit picture in picture
+    exitPictureInPicture: 'exitPictureInPicture',
+    // picture in picture element
+    pictureInPictureElement: 'pictureInPictureElement',
+    // picture in picture enabled
+    pictureInPictureEnabled: 'pictureInPictureEnabled',
+  },
+] as const;
+
+const getPictureInPictureMethods = (element?: HTMLMediaElement | null) => {
+  if (isClient()) {
+    for (const methods of methodsList) {
+      const exitPictureInPicturenMethod = methods.exitPictureInPicture;
+
+      if (exitPictureInPicturenMethod in document) {
+        return { methods, element: element as HTMLVideoElement };
+      }
+    }
+  }
+
+  return { methods: null };
+};
+
+export const addPictureInPictureEventListener = (
+  callback: EventListenerOrEventListenerObject,
+) => {
+  const { methods } = getPictureInPictureMethods();
+
+  if (methods) {
+    document.addEventListener(methods.exitPictureInPicture, callback, false);
```

Comment:
I don't think this is an event
https://developer.mozilla.org/en-US/docs/Web/API/Picture-in-Picture_API#events

Reply:
Yep, `exitPictureInPictureMode` is a separate API: https://developer.mozilla.org/en-US/docs/Web/API/Document/exitPictureInPicture

Reply:
Hmm but we're adding an event listener here and that event name doesn't exist

Reply:
I see, then I believe it is better to use `leavepictureinpicture`

https://developer.mozilla.org/en-US/docs/Web/API/HTMLVideoElement/leavepictureinpicture_event